staramr/blast/results/plasmidfinder/PlasmidfinderHitHSP.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import re
+
+from staramr.blast.results.AMRHitHSP import AMRHitHSP
+
+logger = logging.getLogger('PlasmidfinderHitHSP')
+
+"""
+A Class storing a PlasmidFinder-specific BLAST hit/HSP.
+"""
+
+class PlasmidfinderHitHSP(AMRHitHSP):
+
+    def __init__(self, file, blast_record):
+        """
+        Builds a new PlasmidfinderHitHSP.
+        :param file: The input file.
+        :param blast_record: The Bio.Blast.Record this hit came from.
+        :param hit: The particular Bio.Blast.Record.Alignment.
+        :param hsp: The particular Bio.Blast.Record.HSP.
+        """
+        super().__init__(file, blast_record)
+
+        logger.debug("record=%s", self._blast_record)
+
+        re_search = list(filter(None, re.split('_', self.get_amr_gene_id())))
+
+        if not re_search:
+            raise Exception("Could not split up seq name for [" + self.get_amr_gene_id() + "]")
+
+        length = len(re_search)
+        self._gene = re_search[0]
+        self._gene_variant = re_search[1]
+
+        if length == 3:
+            self._accession = re_search[2]
+        elif length == 4:
+            self._accession = re_search[3]
+        elif length == 5:
```

Comment:
Whenever I have an unhandeled `else` case which could result in some invalid state I normally raise an exception with a descriptive message. That way, if this case is ever encountered the program will crash and say exactly why. If someone encounters this case, they would (ideally) report a bug to us and we can fix it and release a new version.

So, could you add checks to make sure `re_search` has the expected lengths and raise exceptions if not? For example, if length < 3 or > 5 you should raise an exception. You can also include the original string we were trying to parse (`self.get_amr_gene_id()`) in the exception, which should provide useful information to debug.

Reply:
Sure thing, I've made a case where if the length isn't between 3 to 5 it would raise an error also an extra case where it would raise an error if it doesn't assign the values properly in 36238aa